### PR TITLE
Add GitHub Actions CI workflow

### LINKED ISSUE #2: Set up CI
We should set up CI and have it being required for merging PR's. Github Actions is a good alternative and one makes the CI required by inspecting the branch protection rules in Settings. If there are no tests in the repository at the time, just make a dummy one that can be replaced later.

### PR DESCRIPTION
**Pull request by haavardnk**
Added CI. Action fails because previous pull request not yet merged.

Questions:
- Which python versions to test? Currently mirrored Config Suite.
- Is it OK to run pytest simply like this, or should I use setup.py testing?
- Requested access to equinor repo from Codacy
- Should all my files also include the Equinor copyright text at the top?

Closes #2

### CODE REVIEW

File: .pylintrc [hunk 1]
```diff
@@ -0,0 +1,439 @@
...
+# can either give multiple identifiers separated by comma (,) or put this
+# option multiple times (only on the command line, not in the configuration
+# file where it should appear only once).You can also use "--disable=all" to
+# disable everything first and then reenable specific checks. For example, if
+# you want to run only the similarities checker, you can use "--disable=all
+# --enable=similarities". If you want to run only the classes checker, but have
+# no Warning level messages displayed, use"--disable=all --enable=classes
+# --disable=W"
+disable=
+    superfluous-parens,
```
Comment (markusdregi) [inline .pylintrc:54]: Try to remove these one-by-one; if there are some you do not validate, feel free to remove them...

**Comment by markusdregi** [comment]
What one could typically do here is to set up the branch to be merged into the one containing the package instead of `main`. That way the tests would pass and you can demonstrate that is passes before the other PR is merged. It is not very often one need this, but every now and then it is convenient to rig the branches that way...

**Python versions**: Test all the ones you can. If you get it to pass for `3.5 - 3.9`, that is perfect. And then one can consider dropping older Python versions to gain newer features in the language. That is often a trade off. Our internal environment is currently 3.6, hoping to get to 3.7+ during the year. So `3.5` support is not worth much in Equinor at least 🤷

**Comment by markusdregi** [comment]
Running `pytest` directly is preferred as some of the tooling for testing via `setup.py` is deprecated.

**Comment by markusdregi** [comment]
> Should all my files also include the Equinor copyright text at the top?

I don't think that is necessary no...

**Comment by haavardnk** [comment]
Removed all unnecessary linting and merged main for tests to pass. Check if OK @markusdregi

**Comment by markusdregi** [comment]
Looks good @haavardnk! Could you try squashing the two commits into one? :)

After that, feel free to merge :+1:

**Review by markusdregi — APPROVED** [review]

**Pull request merged**